src/components/Pricing/test/PlanComparisonTest/index.tsx

```diff
@@ -0,0 +1,457 @@
+                }
+
+                return (
+                    <div key={product.name + '-tiers-' + i} className="pr-4">
+                        {parseFloat(tier.unit_amount_usd) === 0 ? (
+                            <div>
+                                <span className="font-bold text-base">Free</span> up to{' '}
+                                {convertLargeNumberToWords(tier.up_to, null, true, product.type)}, then
+                            </div>
+                        ) : (
+                            <>
+                                <span className="font-bold text-base">${parseFloat(tier.unit_amount_usd)}</span>
+                                {/* the product types we have are plural, so we need to singularlize them and this works for now */}
+                                <span className="text-gray">/{product.type.replace(/s$/, '')}</span>
+                                <p className="text-sm mb-0">
+                                    <Link to="" onClick={() => setModalOpen(true)} className="text-red font-bold">
+                                        Volume discounts
+                                    </Link>{' '}
+                                    after {convertLargeNumberToWords(tier.up_to)}/mo
+                                    <ProductTiersModal
+                                        modalOpen={modalOpen}
+                                        setModalOpen={setModalOpen}
+                                        product={product}
+                                    />
+                                </p>
+                            </>
+                        )}
+                    </div>
+                )
+            })}
+        </div>
+    )
+}
+
+const icons = {
+    product_analytics: ProductIcons.analytics,
+    session_recording: ProductIcons.sessionRecording,
+    feature_flags: ProductIcons.featureFlags,
+    experiments: ProductIcons.experiments,
+    integrations: ProductIcons.appLibrary,
+    platform: NotProductIcons.platform,
+    support: NotProductIcons.support,
+}
+
+export const getProductLimit = (product?: BillingProductV2Type): JSX.Element => {
+    if (!product) return <></>
+    if (!product.free_allocation) {
+        return <span className="font-bold">Unlimited</span>
+    }
+    return (
+        <span className="font-bold">
+            {convertLargeNumberToWords(product.free_allocation)} {product.type}
+            <span className="font-normal text-black/50">/mo</span>
+        </span>
+    )
+}
+
+export const PlanComparisonTest = ({ className = '' }) => {
+    const posthog = usePostHog()
+    const [availablePlans, setAvailablePlans] = useState<BillingV2PlanType[]>([])
+
+    const excludedFeatures = [
+        'dashboard_collaboration',
+        'ingestion_taxonomy',
+        'terms_and_conditions',
+        'security_assessment',
+        'app_metrics',
+        'paths_advanced',
+    ]
+
+    useEffect(() => {
+        const fetchPlans = async () => {
```

Comment:
I'd really like to see this data fetched at build time to catch potential errors before they're shown to the user. Would also enable us to remove the loading spinner and show the data immediately. Not mention fewer client HTTP requests! Any reason we _shouldn't_  do that?

Reply:
In the future we will be running experiments here so we don't know what the plans will be at build time (I don't have it set up in this way yet but will soon).

Reply:
Got it!